encode.go

```diff
@@ -1345,6 +1362,14 @@ func encodeMarshalerType(e *encoderBuffer, em *encMode, v reflect.Value) error {
 	if err != nil {
 		return err
 	}
+
+	// Verify returned CBOR data item from MarshalCBOR() is well-formed and passes tag validity for builtin tags 0-3.
+	d := decoder{data: data, dm: defaultDecMode}
```

Comment:
Does it make sense to configure the nesting depth and array/map length on this decoder to their maximums to avoid rejecting Marshaler output that would have been the allowable output of a big non-Marshaler object? And should Marshalers be able to produce indefinite-length or tagged items if those are disabled in the EncMode?

Reply:
Good catch!  Thank you!  Done in https://github.com/fxamacker/cbor/pull/485/commits/a3a1d711f1981bcfa161a754fae2d2bf55d4b351.